src/fstab-generator/fstab-generator.c

```diff
@@ -1297,7 +1297,7 @@ static int determine_usr(void) {
  * with /sysroot/etc/fstab available, and then we can write additional units based
  * on that file. */
 static int run_generator(void) {
-        int r, r2 = 0, r3 = 0;
+        int r, ret = 0;
```

Comment:
Frankly, I'm not sure that "simplify" is the right word. The new code seems much more complex ;)

Reply:
OK. I will update the commit message.